File: MtgCoreLib/Commands/PropertyFilter.cs
using System.Collections.Generic;

public class PropertyFilter<T> {
    public string Property { get; set; }
    public PropertyFilterOperator Operator { get; set; }
    public object Value { get; set; }
}

public enum PropertyFilterOperator {
    Contains,
    IsEqual,
    IsContainedIn,
}

public static class PropertyFilterOperatorHelper {
    public static bool TryParse(string s, out PropertyFilterOperator propertyFilterOperator) {
        propertyFilterOperator = PropertyFilterOperator.IsEqual;

        switch (s) {
            case "eq":
                propertyFilterOperator = PropertyFilterOperator.IsEqual;
                break;
            case "con":
                propertyFilterOperator = PropertyFilterOperator.Contains;
                break;
            case "in":
                propertyFilterOperator = PropertyFilterOperator.IsContainedIn;
                break;
            default:
                return false;
        }

        return true;
    }
}

File: frontend/src/app/shared/filtering/property-filter.ts
import { QueryStringGenerator } from '../utils/query-string-generator.interface';
import { PropertyFilterOperator, PropertyFilterOperatorExtensions } from './property-filter-operator';

export class PropertyFilter implements QueryStringGenerator {
  property: string;
  operator: PropertyFilterOperator;
  value: any;

  constructor(init?: Partial<PropertyFilter>) {
    Object.assign(this, init);
  }

  toQueryString(): string {
    return this.property + ' ' + PropertyFilterOperatorExtensions.toQueryStringOperator(this.operator) + ' ' + this.valueQueryString();
  }

  private valueQueryString(): string {
    switch (this.operator) {
      case PropertyFilterOperator.Contains:
      case PropertyFilterOperator.IsEqual:
        return '\'' + this.value + '\'';
      case PropertyFilterOperator.IsContainedIn:
        if (this.value instanceof Array) {
          return '[\'' + this.value.join('\',\'') + '\']';
        }
    }
  }
}
